### ISSUE TITLE
Does not compile on HP-UX at all

### ISSUE THREAD

**Issue by michael-o** [stedolan/jq#779]
Downloaded 1.4 tarball, ran configure. Running `gmake` gives me:

    ~/jq-1.4$ gmake
      GEN      version.h
    gmake  all-am
    gmake[1]: Entering directory '/net/home/oracle/jq-1.4'
      CC       locfile.lo
    cc: error 1914: bad form for `-W' option
    Makefile:821: recipe for target 'locfile.lo' failed
    gmake[1]: *** [locfile.lo] Error 1
    gmake[1]: Leaving directory '/net/home/oracle/jq-1.4'
    Makefile:648: recipe for target 'all' failed
    gmake: *** [all] Error 2

**Comment by wtlangford** [comment]
Can we get your configure.log, please?

**Comment by michael-o** [comment]
Yes, I will prepare everything you need. Unfortunately, I cannot upload any files here.

**Comment by michael-o** [comment]
[Here](http://pastebin.com/248Six7v) is the output of configure and [here](http://pastebin.com/18f1KZTk) the of config.log.

**Comment by wtlangford** [comment]
The issue here is that your compiler is HP's C/aC++, not gcc.  That being said, the configure script just looks for a cc, not gcc.  I don't know that we actually intended to -only- support gcc/llvm, though.  A quick fix for your issue could be to remove the `-Wextra -Wall -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-function` values from AM_CFLAGS in Makefile.am and then re-run `./configure`.  It looks like your compiler is just mad about those flags.  You don't need them to build, they're just diagnostic for when we work.

**Comment by wtlangford** [comment]
@nicowilliams we should look into tweaking the automake files so that they only add the -W flags when compiling under gcc or something gcc-like (llvm).

**Comment by michael-o** [comment]
@wtlangford, I will have a look at it and give you feedback. You can have a look at curl's autoconf files. They are working fine on HP-UX. Moreover, I can provide compatible parameters for HP's compiler.

**Comment by michael-o** [comment]
Just reran `autoreconf -i` seems to require Git but passes. I removed the `AM_CFLAGS`.

    $ gmake
      GEN      version.h
    gmake  all-am
    gmake[1]: Entering directory '/net/home/osipovmi/jq-1.4'
      CC       locfile.lo
      CC       bytecode.lo
      CC       compile.lo
      CC       execute.lo
    "execute.c", line 54: error #2094: the size of an array must be greater than
              zero
        union frame_entry entries[0];
                                  ^
    
    1 error detected in the compilation of "execute.c".
    Makefile:821: recipe for target 'execute.lo' failed
    gmake[1]: *** [execute.lo] Error 1
    gmake[1]: Leaving directory '/net/home/osipovmi/jq-1.4'
    Makefile:648: recipe for target 'all' failed
    gmake: *** [all] Error 2

**Comment by michael-o** [comment]
The Git problem comes from line 1 in `configure.ac`:

     m4_define([jq_version], m4_esyscmd_s([git describe --tags --dirty --match 'jq-*'|sed 's/^jq-//']))

This does not make sense if someone builds from a source tarball.

**Comment by michael-o** [comment]
I made some progess, error 2094 can be resolved with `CFLAGS=-Agcc`. Maybe this is not the best approach. Here is the next problem:

    $ gmake
    gmake  all-am
    gmake[1]: Entering directory '/net/home/osipovmi/jq-1.4'
      CCLD     libjq.la
    cc: warning 901: unknown option: `-hared': use +help for online documentation.
    cc: warning 983: The -lc library specified on the command line is also added automatically by the compiler driver.
    ld: Unsatisfied symbol "main" in file <no file>
    1 error.
    Makefile:725: recipe for target 'libjq.la' failed
    gmake[1]: *** [libjq.la] Error 1
    gmake[1]: Leaving directory '/net/home/osipovmi/jq-1.4'
    Makefile:648: recipe for target 'all' failed
    gmake: *** [all] Error 2

**Comment by michael-o** [comment]
Friends, I have made it work:

    $ uname -a && ~/jq/bin/jq . upload-response-7.json
    HP-UX <hostname> B.11.31 U ia64 HP-UX
    {
      "uid": "hFYVom8bBv4vwA"
    }
 
The error in the previous comment is caused by the flawed `libtool`. I have replaced it with the libtool generated on the same machine for `curl` and I have now a working binary.

`gmake check` fails because `valgrind` is not installed.
